test/e2e/selectors/selectors_test.go

```diff
@@ -0,0 +1,274 @@
+package selectors
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	"github.com/flightctl/flightctl/test/login"
+	"github.com/flightctl/flightctl/test/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	deviceYAMLPath  = "examples/device.yaml"
+	fleetYAMLPath   = "examples/fleet.yaml"
+	repoYAMLPath    = "examples/repository-flightctl.yaml"
+	unknownSelector = "unknown or unsupported selector"
+	resourceCreated = `(200 OK|201 Created)`
+	unknownFlag     = "unknown flag"
+	failedToParse   = "failed to parse"
+)
+
+// Define the struct for parsing device.yaml
+type Device struct {
+	Metadata struct {
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
+
+func TestFieldSelectors(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Field Selectors E2E Suite")
+}
+
+// Utility function to extract the metadata.name from a YAML file
+func extractNameFromYAML(filePath string) (string, error) {
```

Comment:
You could reuse the method I added: GetDeviceByYaml
and get the device name by *device.Metadata.Name
See the example in cli_test